### PR TITLE
Reduce the global `codespell` ignore words list

### PR DESCRIPTION
**Pull request by eerovaher**
### Description

It was possible to remove many words from the global `codespell` ignore list with very little effort. In a handful of cases it was reasonable to replace a global ignore with a per-line ignore. It was also useful to configure `codespell` to ignore the automatically generated `*_parsetab.py` and `*_lextab.py` files.

There is now less clutter in our `pyproject.toml` and a few previously ignored misspellings will now be prevented from entering our codebase.

There's a few more entries in the global ignore list that could be removed, but removing them requires a bit more effort, so it would be best to address them separately.

- [x] By checking this box, the PR author has requested that maintainers do **NOT** use the "Squash and Merge" button. Maintainers should respect this when possible; however, the final decision is at the discretion of the maintainer that merges the PR.

### Approvals by sub-package

- [x] `coordinates` https://github.com/astropy/astropy/pull/16736#pullrequestreview-2182542464
- [x] `cosmology` https://github.com/astropy/astropy/pull/16736#pullrequestreview-2181394351
- [x] `io.ascii` https://github.com/astropy/astropy/pull/16736#pullrequestreview-2182777095
- [x] `io.fits` https://github.com/astropy/astropy/pull/16736#discussion_r1693434150
- [x] `io.misc`
- [x] `nddata` https://github.com/astropy/astropy/pull/16736#pullrequestreview-2181163219
- [x] `time` https://github.com/astropy/astropy/pull/16736#pullrequestreview-2182777095
- [x] `units` https://github.com/astropy/astropy/pull/16736#pullrequestreview-2182542464
- [x] `wcs`

### CODE REVIEW

**Review by mwcraig — APPROVED** [review]

File: astropy/io/fits/card.py
Comment (pllim) [inline astropy/io/fits/card.py:1]: This changed real code?

**Review by nstarman — APPROVED** [review]
G2G 4 Cosmology

**Review by mhvk — APPROVED** [review]
Nice, good to just make things more readable! Explicit approval for coordinates and units, though everything looks fine to me.

**Review by taldcroft — APPROVED** [review]
Good for io.ascii, time.

File: astropy/io/fits/card.py
Comment (eerovaher) [reply astropy/io/fits/card.py:1]: Yes.

File: astropy/io/fits/hdu/table.py
Comment (eerovaher) [reply astropy/io/fits/hdu/table.py:1]: Why not?

File: astropy/io/fits/card.py
Comment (eerovaher) [reply astropy/io/fits/card.py:1]: `codespell` doesn't even attempt to figure out what is a variable name and what isn't, it just looks for words that look like mispellings.

File: astropy/io/fits/hdu/table.py
Comment (eerovaher) [reply astropy/io/fits/hdu/table.py:1]: `codespell` thinks that the variable name "dout" is a misspelling of "doubt", but instead of renaming the variable it was better to remove it entirely.

File: astropy/io/fits/card.py
Comment (eerovaher) [reply astropy/io/fits/card.py:1]: It's not changing the regex pattern, it's changing the name of a [named group](https://docs.python.org/3/howto/regex.html#non-capturing-and-named-groups).

**Comment by eerovaher** [comment]
@dhomeier, as a general package maintainer you should have the authority to approve the changes in any of the sub-packages. Could you mark in the checklist of the opening post which packages you actually approved for?

**Review by dhomeier — CHANGES_REQUESTED** [review]
@dhomeier, as a general package maintainer you should have the authority to approve the changes in any of the sub-packages. Could you mark in the checklist of the opening post which packages you actually approved for?

Since Pey Lian already tagged @saimn I am not approving for `io.fits` yet before giving him a few days notice to respond.

File: astropy/time/tests/test_basic.py
Comment (eerovaher) [reply astropy/time/tests/test_basic.py:855]: This would not fit in the 88 character line length limit and would therefore force Ruff to split this over several lines. However, the following is short enough and also avoids using `'` in the string:
```suggestion
                "2000-01-01 01:01", format="iso", scale="tai", in_subfmt="invalid"
```
But I'm not going to make that change unless @taldcroft confirms that I should.

Comment (dhomeier) [reply astropy/time/tests/test_basic.py:855]: Good point. "nonexistent" should also fit.

File: astropy/io/fits/card.py
Comment (eerovaher) [reply astropy/io/fits/card.py:1]: As far as I can tell the only changes that have not been approved are in this file, so if this isn't reviewed by Friday then I'll drop the changes here (i.e. the changes regarding "valu") from the pull request so that the rest of the changes could be merged.

Comment (mhvk) [reply astropy/io/fits/card.py:1]: FWIW, I think it is a clear change for the better...

Comment (saimn) [reply astropy/io/fits/card.py:1]: The change looks fine, though I'm not sure it's worth the effort and time... and given the PR title I did not expect to have a look at this one.

**Comment by eerovaher** [comment]
The changes to `io.fits` have been approved. There was also a question about `time`, but my opinion regarding that is the same as it was when I wrote https://github.com/astropy/astropy/pull/16736#discussion_r1684995571. This pull request can now therefore be merged as is.

**Comment by mhvk** [comment]
@dhomeier - you quested changes here. Could you check whether you are happy with what was done, so we can merge this?

**Comment by dhomeier** [comment]
It's not clear to if the [suggested change](https://github.com/astropy/astropy/pull/16736#discussion_r1684995571) is going to to be applied now, since I have not seen approval from @taldcroft – my concern that introducing a non-standard character like `'` _might_ raise a spurious exception, even if not very likely, still stands (besides, it is not much of an improvement to written English imo ;-)

File: astropy/time/tests/test_basic.py
Comment (dhomeier) [reply astropy/time/tests/test_basic.py:855]: For the record, I was not arguing for improving or extending the existing (previous) test, just would prefer not to weaken it by introducing a non-alphanumerical character to the string only for the sake of keeping a spellchecker happy about a variable _name_. But since @taldcroft has not commented since, I am willing to let his initial approval stand.

Comment (eerovaher) [reply astropy/time/tests/test_basic.py:855]: Raising the question was reasonable enough, but if the sub-package maintainer doesn't think it should be addressed then this should be merged as is.

**Pull request merged**
